elements/lisk-p2p/src/peer_selection.ts

```diff
@@ -146,15 +146,15 @@ export const selectPeersForConnection = (
 		if (shuffledTriedPeers.length !== 0) {
 			if (Math.random() < r) {
 				// With probability r
-				return shuffledTriedPeers.splice(0, 1)[0];
+				return shuffledTriedPeers.pop() as P2PPeerInfo;
 			}
 		}
 
 		if (shuffledNewPeers.length !== 0) {
 			// With probability 1-r
-			return shuffledNewPeers.splice(0, 1)[0];
+			return shuffledNewPeers.pop() as P2PPeerInfo;
 		}
 
-		return shuffledTriedPeers.splice(0, 1)[0];
+		return shuffledTriedPeers.pop() as P2PPeerInfo;
```

Comment:
This one could be undefined if the `shuffledTriedPeers` length is 0.

Reply:
Or if `shuffledNewPeers` is 0 and the `if (Math.random() < r)` block is not executed for `shuffledTriedPeers`.

Reply:
Those two cases are covered above. Have a look!